--- a/include/llvm/ADT/IntrusiveRefCntPtr.h	2017-12-20 10:19:54.046086400 -0800
+++ b/include/llvm/ADT/IntrusiveRefCntPtr.h	2017-12-20 10:19:40.949776900 -0800
@@ -197,6 +197,9 @@
   private:
     void retain() { if (Obj) IntrusiveRefCntPtrInfo<T>::retain(Obj); }
     void release() { if (Obj) IntrusiveRefCntPtrInfo<T>::release(Obj); }
+
+    template <typename X>
+    friend class IntrusiveRefCntPtr;
   };
 
   template<class T, class U>
